### ISSUE TITLE
File attachments

### ISSUE THREAD

**Issue by Strajk** [HappenApps/Quiver#83]
I love using Quiver as personal notebook for all the things as a replacement for Evernote, and I miss attaching files to notes..., like PDFs, Word docs, small archives and so on...

I know that PDF and Word docs are nasty things to put into programmer's notebook, but world is is sometimes not a nice place :))

What do you think about this? Would it be do-able? :)

**Comment by ylian** [comment]
Could you try the build in https://github.com/HappenApps/Quiver/issues/3?

If you import Evernote notes with attachments, the importer will copy over the file, and create a file link. Let me know if the file link is an okay solution. That will provide a quick implementation for this feature.

Eventually I need to implement a UI like Evernote does.

![screen shot 2015-03-29 at 10 37 11 am](https://cloud.githubusercontent.com/assets/984895/6886067/964ad768-d5ff-11e4-9639-2821d4ade522.png)

**Comment by scott-mead** [comment]
I'd like to be able to embed attachments in to notes. This would simplify team sharing quite a bit.

**Comment by ylian** [comment]
Almost there.

**Comment by LeonardoGentile** [comment]
will this be available in the incoming release?

**Comment by ylian** [comment]
No, there are some issues with my implementation. Have to go back to the drawing board.

**Comment by asafge** [comment]
+1

**Comment by RoCry** [comment]
+1

I will need this feature to put related files, like iThoughX or even a binary file for sharing to others.

Just simply have a icon/link to the file and place the file with the resources of the note is good enough.

**Comment by scott-mead** [comment]
+1
  Sharing with the team is critical for me. 

## 

Scott Mead via mobile
IPhone : +1-607-765-1395
Skype  : scottm.openscg
Gtalk    : scottm@openscg.com

**Comment by ylian** [comment]
I can implement the simplified version right now --- just create a link to the file and copy the file to the resources folder. The fancy presentation as in Evernote can be implemented later.

**Comment by LeonardoGentile** [comment]
it would be nice for me if you could implement the simplified version to get started.
:+1:

**Comment by ylian** [comment]
A basic implementation of file attachment has been implemented in Quiver 3.0 beta 10: https://github.com/HappenApps/Quiver/issues/162#issuecomment-150564507

When you drag or paste a file into a text cell, you will see a prompt like this:

![screen shot 2015-10-23 at 8 38 14 am](https://cloud.githubusercontent.com/assets/984895/10693278/3ab098b6-7965-11e5-8276-b7b141ae4279.png)

So you can choose to copy the file to the note or just create a reference link. Let me know if the wording can be improved.

You can also use the "Attach" button in the toolbar (the one looking like a paper clip). The attach button works in Markdown cells too.

**Comment by LeonardoGentile** [comment]
@ylian I've tried to copy a file, then deleting it or deleting the cell containing it. 
I've opened the note in finder and the file is still there. Should't we delete it or are you making some safe things (like trash or so)?

**Comment by ylian** [comment]
@LeonardoGentile Can you double check? I tried this several times from text cells and Markdown cells, and the file is deleted.

Note that resources (images, files) are now named by MD5 of their content, for de-duplication. So if you have another reference of the same file in the note, the file isn't deleted unless you delete all the references.

**Comment by X-G** [comment]
Can I copy the files in the notebook out to any folder in my computer? I tried but didn't find the solution.
@ylian

**Comment by ylian** [comment]
@X-G Right now this implementation is quite basic. So the only way to do this is to right click on the note from the list, "Show in Finder", go to the "resources" folder, and find the file.

**Comment by JMichaelTX** [comment]
> When you drag or paste a file into a text cell, you will see a prompt like this:

@ylian, FYI, this copy/link prompt is often hidden behind another window (like the Finder), and in a different monitor screen than the one with Quiver.
It would be nice if the prompt was on same monitor as Quiver, and always brought to front.

Thanks.

**Comment by ylian** [comment]
@JMichaelTX Thanks.

**Comment by mkyt** [comment]
It would be great if I could link local resources (e.g. images, videos, etc) from within markdown cells.
For now, I use text cell only to display embedded images, which is quite cumbersome.
